### DISCUSSION TITLE
Livewire\Features\SupportBrowserHistory: parse_str(): Passing null to parameter #1 ($string) of type string is deprecated

### DISCUSSION THREAD

**Discussion by ralphjsmit** [livewire/livewire#4532] [7. Bugs (archived)] [answered]
I'm working through some Honeybadger errors on our production and staging environment and I noticed that the following error appeared several times:

*parse_str(): Passing null to parameter #1 ($string) of type string is deprecated.*

Judging by the backtrace, this error appears to happen in the `getQueryParamsFromRefererHeader()` function:

```php
    public function getQueryParamsFromRefererHeader()
    {
        if (empty($referer = request()->header('Referer'))) return [];

        parse_str((string) parse_url($referer, PHP_URL_QUERY), $refererQueryString);

        return $refererQueryString;
    }
```

The environment is on PHP 8.1. 

I don't really know what the cause could be here, because it looks like you're already checking whether there actually are Referer params. I still wanted to share it though.

I can provide a full backtrace if you want, please let me know!

**Comment by joshhanley** [comment, marked as answer]
@ralphjsmit thanks for reporting! This has been fixed in PR #4376 and should have already been released in Livewire 2.9.0. Can you upgrade to that and see if it fixes the issue for you?

Thanks!

**Reply by ralphjsmit** [reply]
Hey Josh, awesome! Will check out the recent PRs next time 😅

**Reply by joshhanley** [reply]
@ralphjsmit heh all good! 🙂
